### DISCUSSION TITLE
Existing HTMLMediaElement with predefined peaks

### DISCUSSION THREAD

**Discussion by mutil** [katspaugh/wavesurfer.js#2894] [Q&A]
Hello and thanks for this wonderful library!

I try to use an existing audio element with peaks calculated in server, but it doesn’t seem to work. Using wavesurfer.js@7.0.0-beta.8, the waveform is not rendered when peaks are provided this way. Am I missing something?

Minimal example to reproduce:
```
<audio src='/test.mp3' preload="metadata" controls></audio>
<div id="wf"></div>

<script type="module">
  import WaveSurfer from 'https://unpkg.com/wavesurfer.js@beta';

  const audio = document.querySelector('audio');
  const wavesurfer = WaveSurfer.create({
      container: '#wf',
      media: audio,
      // url: audio.src,
      peaks: [[0, 0.1, 0.2, 0.3]],
   });
</script>
```
If `url` option is provided instead of `media`, or `peaks` option is omitted, waveform renders correctly.
Also, wavesurfer events (e.g. `ready`) won’t fire when using `media` and `peaks` option together.

Tested in firefox and chrome, linux.

**Comment by katspaugh** [comment]
I've reproduced it, looks like a bug indeed! Thanks for reporting.

**Comment by MasteringBOX** [comment]
Following up on this. I have the same issue. If url is not provided the peaks are not rendered. If the url is provided along with the peaks, the waveform will only be rendered when the audio is downloaded. The URL parameter is set as optional but it is required for wavesurfer to work.

Ideally, if the peaks are provided, the waveform should display inmediately and the audio should be streamed using the default html5 audio features. I don't mind if this is by including the media element in the options or with just the url param.

**Reply by katspaugh** [reply]
Good point, 100% agree. Created https://github.com/katspaugh/wavesurfer.js/issues/3282 to track this.

**Reply by MasteringBOX** [reply]
Thanks! Would be awesome if this gets addressed.

**Reply by katspaugh** [reply]
Released this in 7.4.0. Please note that you must provide both peaks and duration for this to work. Cheers!
